alibi/explainers/tests/test_simiarlity/test_grad_methods_integration.py

```diff
@@ -211,5 +211,123 @@ def test_grad_cos_result_order_tf(seed):
     )
     explainer = explainer.fit(ds, ds)
     explanation = explainer.explain(ds[0], Y=ds[0])
-    assert (ds[explanation['ordered_indices'][1]] == ds[1]).all()
-    assert (ds[explanation['ordered_indices'][-1]] == ds[-1]).all()
+    assert (ds[explanation['ordered_indices'][0][1]] == ds[1]).all()
+    assert (ds[explanation['ordered_indices'][0][-1]] == ds[-1]).all()
+
+
+def test_multiple_test_instances_grad_cos():
```

Comment:
Is there a reason why we do not combine `test_multiple_test_instances_grad_cos` and `test_multiple_test_instances_stored_grads_grad_cos` in a single test so we avoid code duplication?
At a first glance it seem like the only difference is `precompute_grads=True`, which you can set it on or of through `pytest.mark.parametrize`. 
Or am I missing something? 
Same applies for the other tests, so I won't repeat myself.